### DISCUSSION TITLE
Itterative solver for a highly non-linear problem

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#24013] [Q&A General] [answered]
Hello everyone,

I am solving a problem which is highly non-linear, and iterative solvers failed due do the convergence. Now I am running my problem with a direct solver, but at a very high computational cost. Is there any suggestion for me to use iterative solver and still get convergence?

Thanks,
Ali

**Comment by GiudGiud** [comment]
Hello

How many variables?
Can you describe the physics involved?

Also could please **attach** the convergence logs for an iterative and a direct solver?
If you could turn on the residual prints with
```
[Debug]
  show_var_residual_norms = true
[]
```

Also are the nonlinear iterations diverging? Or the linear iterations? Using a direct solver typically mostly influences the latter

Guillaume

**Reply by alimostafavi24** [reply]
I can do that for future simulations. In my simulation, phase field and navier stokes are fully coupled.

```
 Elems:                   
    Total:                 2250
    Local:                 24
    Min/Max/Avg:           23/24/23
  Num Subdomains:          1
  Num Partitions:          94
  Partitioner:             metis

Nonlinear System:
  Num DOFs:                11780
  Num Local DOFs:          205
  Variables:               { "vel_x" "vel_y" "p" "c" "w" } 
  Finite Element Types:    "LAGRANGE" 
  Approximation Orders:    "FIRST" 
```

**Reply by GiudGiud** [reply]
Can you drop pressure to zero-th order?
In incompressible flow it acts as a Lagrange multiplier and the numbers of degrees of freedom on this are best limited

**Reply by lindsayad** [reply, marked as answer]
I don't actually think that zeroth order pressure is stable. Trying it out on a test problem it yields a singular matrix. If you wanted to do constant monomial for pressure you'd likely need to add discontinuous galerkin face flux terms for the mass equation.

But for the variables used here, @alimostafavi24, you do have to make sure you're using pressure stabilization if you're going to do equal order discretization with velocity. I don't know if you're already doing that?

If you want something other than LU for Navier-Stokes, you're probably going to want to explore field split options. You can look at some discussion on https://github.com/idaholab/moose/pull/22405.
